src/server/oasisapi/files/views.py

```diff
@@ -39,19 +48,33 @@ def _handle_get_related_file(parent, field, file_format):
     # Parquet format requested and data stored as csv
     if file_format == 'parquet' and f.content_type == 'text/csv':
         output_buffer = io.BytesIO()
-        df = ods_tools.read_csv(io.BytesIO(f.file.read()))
+
+        # Load DataFrame and pass to ods-tools exposure class
+        exposure = OedExposure(**{
+            EXPOSURE_ARGS[field]: pd.read_csv(io.BytesIO(f.file.read()))
```

Comment:
I kind of wish we could just put 
EXPOSURE_ARGS[field]: f.file
what type is f.file.
I may be able to add it to ods_tools supported format

Reply:
Having an option to read from a binary stream (like `io.BytesIO`) would be handy, I think that task should belong with ods-tools. 

Currently the platform loads the bytes into pandas DataFrame before passing over to the Exposure class, which is there as a work around. 

The `f.file` object itself from Django, and is a wrapper around the file storage classed loaded in Django's settings. 
For example, it could be a file from disk, S3 Object or Azure Blob...  etc  

```
> f     
<RelatedFile: File_d9542031640d42e2b7f27560be2290d5.csv>
> f.file
<FieldFile: d9542031640d42e2b7f27560be2290d5.csv>
> type(f.file)
<class 'django.db.models.fields.files.FieldFile'>
``` 
* `RelatedFile` is from here https://github.com/OasisLMF/OasisPlatform/blob/develop/src/server/oasisapi/files/models.py#L57-L69

* `FieldFile` is from Django Storage

Reply:
I agree it'd be nice if `ods_tools` supported this natively.
For this PR on Platform, I approve the current implementation with `io.BytesIO` workaround - we'll update if/when we update the `ods_tools` API.

Looking at `OedExposure`, I see there is an `OedSource.from_file_path()` method that is triggered if the file is a path, but `from_file_path` only returns an initialised `OedSource` without actually reading the data from file. 
Since `OedSource.from_dataframe()` loads the data, I think we should just have `OedSource.from_file_path()` to load the data before returning.
@sstruzik can we do that or perhaps open an issue on the ods repo to track this?